### ISSUE TITLE
Lack of DRONE_VERSION

### ISSUE THREAD

**Issue by k0nserv** [athieriot/drone-artifactory#21]
So I think the `DRONE_VERSION` environment variable was removed from Drone at some point between 0.5 and 0.8. This causes the plugin to go down the pre 0.5 code path and fail despite the fixes in #19 and #20. I've asked in the issue that requested the variable be introduced. https://github.com/drone/drone/issues/1415

**Comment by athieriot** [comment]
Oh man.

Maybe there is an another way to detect the version. Hopefully.

**Comment by k0nserv** [comment]
Yeah I'm not sure. Another option is to publish tagged versions for the plugin with different support and elliminate the 0.4 support. However there will be other users depending on `latest` being compatible with 0.4 so that's not a good idea as it stands

**Comment by athieriot** [comment]
Let's wait for their answers and I can push a 0.4 tag along.

I don't think it's too harsh to make latest going further and be, as its
name suggest, the latest (Considering they are already version 0.8)

**Comment by k0nserv** [comment]
Well there aren't any tags so people didn't previously have the option to depend on a specific version. I know a lot of our usage would break with such a change

**Comment by tboerger** [comment]
Just to note, everything <= 0.5 is EOL on the drone core side.

**Comment by Poogles** [comment]
Can't we just look for version unique env variables?  Looking at [latest](http://docs.drone.io/environment-reference/) and [0.5](http://readme.drone.io/0.5/usage/environment-reference/) vs [0.4](http://readme.drone.io/0.4/usage/variables/) then `DRONE_PREV_COMMIT_SHA` might be an option?

**Comment by k0nserv** [comment]
It turns out you can do this

```yaml
    environment:
      - DRONE_VERSION=0.8 # Attempt to fool the artifactory plugin
```

**Comment by athieriot** [comment]
Do you like that or... ?

I still think we could benefit from having Docker tags anyway

**Comment by k0nserv** [comment]
It works on latest to make `drone-artifactory` work with Drone 0.8. We could definitely benefit from tags, however making breaking changes to `latest` would break a lot of peoples deployments. I know that internally we'll have a lot of projects that use 0.4 with `drone-artifactory:latest`. These would all break if changes that aren't compatible with Drone 0.4 are pushed to `latest`

**Comment by athieriot** [comment]
I think I feel a bit weird of using an env variable (I mean, that the user
has to set up) in order to toggle Drone implementation.

As a user yourself, do you see that as an issue?

**Comment by k0nserv** [comment]
It's definitely a workaround, however maintaining Drone 0.4 compatibility is important.

**Comment by athieriot** [comment]
Maybe we could keep `latest` as having 0.4 support and new tags for the
following?

**Comment by tboerger** [comment]
Keeping a dead version is important?

**Comment by k0nserv** [comment]
It would be if tagged version had been published previously and encouraged. Given that that's not the case, yes it's important as it would break usage for anyone with < Drone 0.5

**Comment by tboerger** [comment]
I don't believe that there are lots of such old installations flying around but it's up to you.
